frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerLifeCycle.java

```diff
@@ -99,6 +102,16 @@ public void startWorker(int port) throws WorkerInitializationException, Interrup
             if (launcherAvailable) {
                 ArrayList<String> args = launcherArgsToList();
                 argl.addAll(args);
+
+                // multi-worker core pinning
+                if (this.numWorker > 1) {
+                    argl.add("--ninstances");
+                    argl.add(String.valueOf(this.numWorker));
+                    argl.add("--instance_idx");
```

Comment:
A WorkerLifeCycle object is associated with one specific backend worker. I can see here each worker is assigned the same idx.

I'm not sure if each backend worker needs a specific instance idx in launcher. Could you confirm that each backend worker can use the same instance idx?

Reply:
Hi @lxning , each worker is assigned `currNumRunningWorkers` https://github.com/min-jean-cho/serve/blob/launcher_core_pinning/frontend/server/src/main/java/org/pytorch/serve/wlm/WorkerLifeCycle.java#L41. For multiple workers, each backend worker needs a specific instance idx in the launcher - otherwise, all workers will be mapped to the same cores. 

Please see below an example of `ninstances` and `instance_idx` for 4 workers. 

```
2022-04-28T11:58:40,347 [DEBUG] W-9000-benchmark_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - Worker cmdline: [/nfs/site/home/minjeanc/anaconda3/envs/py36/bin/python3, -m, intel_extension_for_pytorch.cpu.launch, --node_id, 0, --ninstances, 4, --instance_idx, 0, /ec/pdx/disks/mlp_lab_home_pool_02/minjeanc/.local/lib/python3.6/site-packages/ts/model_service_worker.py, --sock-type, unix, --sock-name, /tmp/.ts.sock.9000]
2022-04-28T11:58:40,331 [DEBUG] W-9001-benchmark_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - Worker cmdline: [/nfs/site/home/minjeanc/anaconda3/envs/py36/bin/python3, -m, intel_extension_for_pytorch.cpu.launch, --node_id, 0, --ninstances, 4, --instance_idx, 1, /ec/pdx/disks/mlp_lab_home_pool_02/minjeanc/.local/lib/python3.6/site-packages/ts/model_service_worker.py, --sock-type, unix, --sock-name, /tmp/.ts.sock.9001]
2022-04-28T11:58:40,324 [DEBUG] W-9002-benchmark_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - Worker cmdline: [/nfs/site/home/minjeanc/anaconda3/envs/py36/bin/python3, -m, intel_extension_for_pytorch.cpu.launch, --node_id, 0, --ninstances, 4, --instance_idx, 2, /ec/pdx/disks/mlp_lab_home_pool_02/minjeanc/.local/lib/python3.6/site-packages/ts/model_service_worker.py, --sock-type, unix, --sock-name, /tmp/.ts.sock.9002]
2022-04-28T11:58:40,344 [DEBUG] W-9003-benchmark_1.0 org.pytorch.serve.wlm.WorkerLifeCycle - Worker cmdline: [/nfs/site/home/minjeanc/anaconda3/envs/py36/bin/python3, -m, intel_extension_for_pytorch.cpu.launch, --node_id, 0, --ninstances, 4, --instance_idx, 3, /ec/pdx/disks/mlp_lab_home_pool_02/minjeanc/.local/lib/python3.6/site-packages/ts/model_service_worker.py, --sock-type, unix, --sock-name, /tmp/.ts.sock.9003]
```